tests/core/middleware/test_transaction_signing.py

```diff
@@ -407,4 +418,200 @@ def test_sign_and_send_raw_middleware_with_byte_addresses(
     raw_txn = actual[1][0]
     actual_method = actual[0]
     assert actual_method == "eth_sendRawTransaction"
-    assert isinstance(raw_txn, bytes)
+    assert is_hexstr(raw_txn)
+
+
+# -- async -- #
+
+
+@pytest_asyncio.fixture
+async def async_result_generator_middleware():
+    return await async_construct_result_generator_middleware(
+        {
+            "eth_sendRawTransaction": lambda *args: args,
+            "net_version": lambda *_: 1,
+            "eth_chainId": lambda *_: "0x02",
+        }
+    )
+
+
+class AsyncDummyProvider(AsyncBaseProvider):
+    async def coro_request(self, method, params):
+        raise NotImplementedError(f"Cannot make request for {method}:{params}")
+
+
+@pytest.fixture
+def async_w3_base():
+    return AsyncWeb3(provider=AsyncDummyProvider(), middlewares=[])
+
+
+@pytest.fixture
+def async_w3_dummy(async_w3_base, async_result_generator_middleware):
+    async_w3_base.middleware_onion.add(async_result_generator_middleware)
+    return async_w3_base
+
+
+@pytest.fixture
+def async_w3():
+    return AsyncWeb3(AsyncEthereumTesterProvider())
+
+
+@pytest_asyncio.fixture
+async def async_fund_account(async_w3):
+    # fund local account
+    accounts = await async_w3.eth.accounts
+    tx_value = async_w3.to_wei(10, "ether")
+    for address in (ADDRESS_1, ADDRESS_2):
+        await async_w3.eth.send_transaction(
+            {"to": address, "from": accounts[0], "gas": 21000, "value": tx_value}
+        )
+        acct_bal = await async_w3.eth.get_balance(address)
+        assert acct_bal == tx_value
+
+
+@pytest.mark.parametrize(
+    "method,key_object,from_,expected",
+    TEST_SIGN_AND_SEND_RAW_MIDDLEWARE_PARAMS,
+)
+@pytest.mark.asyncio
+async def test_async_sign_and_send_raw_middleware(
+    async_w3_dummy,
+    method,
+    from_,
+    expected,
+    key_object,
+):
+    async_w3_dummy.middleware_onion.add(
+        await async_construct_sign_and_send_raw_middleware(key_object)
+    )
+
+    legacy_transaction = {
+        "to": "0x7E5F4552091A69125d5DfCb7b8C2659029395Bdf",
+        "from": from_,
+        "gas": 21000,
+        "gasPrice": 10**9,
+        "value": 1,
+        "nonce": 0,
+    }
+    if isinstance(expected, type) and issubclass(expected, Exception):
```

Comment:
Exceptions could be covered in a separate parameterized test, but really your preference here.

Reply:
Agreed, this is not ideal for new tests. I wanted to just copy the older sync cases so they can share the same test params but revamping both would be a good idea.